src/benchmarks/micro/libraries/System.Net.Http/Perf.WinHttpHandler.cs

```diff
@@ -0,0 +1,159 @@
+                $"Content-Length: {responseLength}\r\n\r\n{new string('a', responseLength)}");
+            ReadOnlyMemory<byte> responseBytes = Encoding.UTF8.GetBytes(responseText);
+
+            _serverTask = Task.Run(async () =>
+            {
+                try
+                {
+                    while (true)
+                    {
+                        using (Socket s = await _listener.AcceptAsync())
+                        {
+                            try
+                            {
+                                Stream stream = new NetworkStream(s);
+                                if (ssl)
+                                {
+                                    var sslStream = new SslStream(stream, false, delegate { return true; });
+                                    await sslStream.AuthenticateAsServerAsync(_serverCert, false, SslProtocols.None, false);
+                                    stream = sslStream;
+                                }
+
+                                using (var reader = new StreamReader(stream, Encoding.ASCII, detectEncodingFromByteOrderMarks: false, bufferSize: 100))
+                                {
+                                    while (true)
+                                    {
+                                        while (!string.IsNullOrEmpty(await reader.ReadLineAsync())) ;
+                                        await stream.WriteAsync(responseBytes);
+                                    }
+                                }
+                            }
+                            catch (SocketException e) when (e.SocketErrorCode == SocketError.ConnectionAborted) { }
+                        }
+                    }
+                }
+                catch { }
+            });
+
+            var ep = (IPEndPoint)_listener.LocalEndPoint;
+            var uri = new Uri($"http{(ssl ? "s" : "")}://{ep.Address}:{ep.Port}/");
+            _handler = new WinHttpHandler();
+            _invoker = new HttpMessageInvoker(_handler);
+
+            if (ssl)
+            {
+                _handler.ServerCertificateValidationCallback = delegate { return true; };
+            }
+
+            _request = new HttpRequestMessage(HttpMethod.Get, uri);
+        }
+
+        [Benchmark]
+        [BenchmarkCategory(Categories.NoAOT)]
+        public async Task Get()
+        {
+            HttpMessageInvoker invoker = _invoker;
+            HttpRequestMessage req = _request;
+
+            using (HttpResponseMessage resp = await invoker.SendAsync(req, CancellationToken.None))
+            using (Stream respStream = await resp.Content.ReadAsStreamAsync())
+            {
+                await respStream.CopyToAsync(Stream.Null);
+            }
+        }
+
+        [Benchmark]
+        [BenchmarkCategory(Categories.NoAOT)]
+        public async Task Get_EnumerateHeaders_Validated()
```

Comment:
I would drop the header benchmarks. The implementation that's being tested is completely separate from `WinHttpHandler` (it'll be the same with `SocketsHttpHandler`).

Reply:
Sounds good, resolved.